cmd/clusterctl/client/cluster/upgrader.go

```diff
@@ -44,10 +44,10 @@ type ProviderUpgrader interface {
 	Plan() ([]UpgradePlan, error)
 
 	// ApplyPlan executes an upgrade following an UpgradePlan generated by clusterctl.
-	ApplyPlan(clusterAPIVersion string) error
+	ApplyPlan(opts InstallOptions, clusterAPIVersion string) error
 
 	// ApplyCustomPlan plan executes an upgrade using the UpgradeItems provided by the user.
-	ApplyCustomPlan(providersToUpgrade ...UpgradeItem) error
+	ApplyCustomPlan(opts InstallOptions, providersToUpgrade ...UpgradeItem) error
```

Comment:
These are breaking API changes. The last time we introduced similar flags to `clusterctl init` we made some breaking changes citing that "We are not currently providing any compatibility guarantees on cluserctl used as a library". Ref: https://github.com/kubernetes-sigs/cluster-api/pull/4934/files#r683272400

@fabriziopandini Can you please confirm if we can still go ahead and make breaking change to the clusterctl library API?